### PR TITLE
a11y improvement: plugins Pagenavigation

### PR DESCRIPTION
**Pull request by zwiastunsw**
### Summary of Changes
The Pagenavigation plugin displays links to the previous and next article. The text of the links are the words "Prev" and "Next". They do not contain any information about the content/topic of the articles. For display users, tooltips with article titles have been added. For Assistive Technology users - aria-label with information that the link gives access to previous/next article and article title.
Developed by Web Accessibility Team.
### Testing Instructions
1. Start the article list page in one category.
2. Open the article from the list.
3. Set the mouse pointer over the links (buttons) Prev and Next. Tooltips should be displayed.
4. Run a screen reader, such as NVDA (or ChromoVox for Chrome). Move tabs between links. Check that the screen reader correctly announces the purpose of the links to previous and next article
### Expected result
1. Tooltips with linked article title are displayed above the Prev and Next buttons.
2. The screen reader will announce links to the titles of the articles to which they lead.
### Actual result
Currently, the user is not informed exactly about the purpose of the link. The content of the links "previous" and "next" is quite vague.
### Documentation Changes Required
1. No changes to the documentation required - no new functionality introduced.
2. You need to add two new language constants do administrator/language/en-GB/en-GB.en-GB.plg_content_pagenavigation.ini file
PLG_PAGENAVIGATION_PREV_A11Y="Access to the previous article: %s"
PLG_PAGENAVIGATION_NEXT_A11Y="Access to the next article: %s"

### CODE REVIEW

**Comment by brianteeman** [comment]
1. Do we really need to add "tooltip"?
2. Please read the correct way to use sprintf https://docs.joomla.org/Formatted_fields_in_language_translation_strings

**Comment by zwiastunsw** [comment]
Thanks, Brian :)
1. Tooltip is not necessary, but improves accessibility for regular users.
2. Right, my mistake!

**Comment by zwiastunsw** [comment]
@brianteeman, can you add this PR to accessibility category?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by ghost** [comment]
@zwiastunsw done

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 3ea8dae1f67ba55cb9c615f9ff952e1af6b39d98<br /><br />I've tested this item successfully (on 3.7.4), it works as described and is a real improvement. 
But I am not sure about the Tooltip. 
It seems to me kind of b/c break - template designers do not expect the tooltips here and have to adapt their design?
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :red_circle: unsuccessfully on 3ea8dae1f67ba55cb9c615f9ff952e1af6b39d98<br /><br />Enable full php error reporting and view the source code you will see the following error
```
<br />
<b>Notice</b>:  Undefined offset: -1 in <b>C:\Apache24\htdocs\cms\plugins\content\pagenavigation\tmpl\default.php</b> on line <b>26</b><br />
<br />
<b>Notice</b>:  Trying to get property of non-object in <b>C:\Apache24\htdocs\cms\plugins\content\pagenavigation\tmpl\default.php</b> on line <b>26</b><br />
PLG_PAGENAVIGATION_NEXT_A11Y```<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by brianteeman** [comment]
Note to replicate that error it must be on a page with ONLY a next link<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by brianteeman** [comment]
the language strings are in the wrong file. If you apply this PR and check it you will see that ;)

The language file you added the strings to is only loaded in the admin. You probably should put them into en-GB.ini in both the /site and /administrator language folders

Also it would be better if they were simply called something like JNEXT_TITLE and JPREVIOUS_TITLE

Finally for accessibility we should not include words such as link to, image to, access to as they are redundant words as many AT will add that text automatically. eg in this example they would announce "link to access to next article: article name"

It is enough to simply say "next article: article name"

So I would suggest the strings should be

JNEXT_TITLE="Next article: %s"
JPREVIOUS_TITLE="Previous article: %s"

**Comment by chmst** [comment]
Language files for plugins are always and only  in the admin language folder?

**Comment by brianteeman** [comment]
@chmst maybe its a different bug then but whatever it is they are not being loaded and I would still put them in a global language file so that they can be reused

**Comment by chmst** [comment]
@Brian, yes, this is a different bug and as I see, you have already opened an issue.

**Comment by brianteeman** [comment]
@chmst no that issue is not related - that one is only a cosmetic issue - nothing to do with this pr

**Comment by zwiastunsw** [comment]
@brianteeman Thank you for your correct suggestions!

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 1f09f3c4e427ab177beef5cd4e7e21b07a9f91c1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

File: plugins/content/pagenavigation/tmpl/default.php [hunk 1]
```diff
@@ -15,16 +15,16 @@
 <?php if ($row->prev) :
 	$direction = $lang->isRtl() ? 'right' : 'left'; ?>
 	<li class="previous">
-		<a href="<?php echo $row->prev; ?>" rel="prev">
-			<?php echo '<span class="icon-chevron-' . $direction . '"></span> ' . $row->prev_label; ?>
+		<a class="hasTooltip" title="<?php echo htmlspecialchars($rows[$location-1]->title); ?>" aria-label="<?php echo JText::sprintf('JPREVIOUS_TITLE', htmlspecialchars($rows[$location-1]->title) ); ?>" href="<?php echo $row->prev; ?>" rel="prev">
```
Comment (Quy) [inline plugins/content/pagenavigation/tmpl/default.php]: Remove space `) );`

File: plugins/content/pagenavigation/tmpl/default.php [hunk 2]
```diff
@@ -15,16 +15,16 @@
...
+			<?php echo '<span class="icon-chevron-' . $direction . '" aria-hidden="true"></span> <span aria-hidden="true">' . $row->prev_label . '<span>'; ?>
 		</a>
 	</li>
 <?php endif; ?>
 <?php if ($row->next) :
 	$direction = $lang->isRtl() ? 'left' : 'right'; ?>
 	<li class="next">
-		<a href="<?php echo $row->next; ?>" rel="next">
-			<?php echo $row->next_label . ' <span class="icon-chevron-' . $direction . '"></span>'; ?>
+		<a class="hasTooltip" title="<?php echo htmlspecialchars($rows[$location+1]->title); ?>" aria-label="<?php echo JText::sprintf('JNEXT_TITLE', htmlspecialchars($rows[$location+1]->title) ); ?>" href="<?php echo $row->next; ?>" rel="next">
```
Comment (Quy) [inline plugins/content/pagenavigation/tmpl/default.php]: Remove space `) );`

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 991dfcc73a5e4adbaf95eed506a36673481293cc<br /><br />retest after cs fixes<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by ghost** [comment]
@chmst can you please retest?

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on 991dfcc73a5e4adbaf95eed506a36673481293cc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17508">issues.joomla.org/tracker/joomla-cms/17508</a>.</sub>

**Comment by wojsmol** [comment]
@franz-wohlkoenig Please set RTC after 2 succesfull tests.

**Comment by ghost** [comment]
RTC after two successful tests.

**Pull request merged**